### ISSUE TITLE
Total Prediction map/summary subsets

### ISSUE THREAD

**Issue by steffilazerte** [psolymos/moosecounter#25]
> For the Total Prediction, if the data are subset then the Map and the Summary show only the subset data but the plot keeps all the data. Could we change it so the plot also has only the subset data?

Added a `ss` argument to allow subsetting the data points in `mc_plot_predfit()` (will be in new PR)

**Comment by steffilazerte** [comment]
A couple of questions: 

1. Would it be worth showing all the map tiles, so the users can still see the full study area, but only 'fill' the tiles in the subset? A bit more finicky, but may be more useful? Or is it better to have the map zoom in by just showing the subset?

2. Is it worth adding the shinyjs package so we can disable the "Download Subset" button when there is no subset actually selected? Right now you can still download the 'subset' and it will be labelled as a subset, but really be the full data. I could avoid this by either disabling this button, or by changing the file name if the subset is the same as the full data.

**Comment by steffilazerte** [comment]
I think this is ready (except for anything to add based on my questions above). 

- We have two download buttons, one for each type of data
- The data also includes an extra row in the Settings sheet for "Subset", either "Full" (for no subset), or something like 
"SURVEY_NAM: Mayo, Mayo Moose Management Unit" if there is a subset.

![image](https://github.com/psolymos/moosecounter/assets/14676081/984b8b67-8aba-4f2c-9052-a7636bc77a4d)

![image](https://github.com/psolymos/moosecounter/assets/14676081/554f86db-42ea-46b7-bb85-a4e044a91cea)

**Comment by psolymos** [comment]
Thanks.

>    Would it be worth showing all the map tiles, so the users can still see the full study area, but only 'fill' the tiles in the subset? A bit more finicky, but may be more useful? Or is it better to have the map zoom in by just showing the subset?

Zooming is better to show the subset in full.

>    Is it worth adding the shinyjs package so we can disable the "Download Subset" button when there is no subset actually selected? Right now you can still download the 'subset' and it will be labelled as a subset, but really be the full data. I could avoid this by either disabling this button, or by changing the file name if the subset is the same as the full data.

I would prefer the shinyjs disable option here. That is the cleanest.

**Comment by steffilazerte** [comment]
Okay, I think we're good to go. 

The download buttons are disabled by default (enabled when there is data), and the subset button is only enabled if a subset is actually selected.

I also added the summary table rownames as a column, so those tables have more context. I think that makes sense?

Finally, I simplified the multilayered validate() calls by using a helper function. So they validate in order date, models, pi, etc. *if* they are provided and ignore them if not (shouldn't look any different, except that now all the messages are consistent, and there's less code in server.R :grin:)
